firmware/port.c

```diff
@@ -442,7 +442,7 @@ ExecStatus port_begin_cmd(PortData *p) {
             sercom_i2c_master_init(p->port->uart_i2c, p->arg[0]);
             pin_mux(p->port->sda);
             pin_mux(p->port->scl);
-            sercom(p->port->uart_i2c)->I2CM.INTENSET.reg = SERCOM_I2CM_INTENSET_SB | SERCOM_I2CM_INTENSET_MB;
+            sercom(p->port->uart_i2c)->I2CM.INTENSET.reg = SERCOM_I2CM_INTENSET_ERROR;
```

Comment:
So here you're enabling registers for errors but postponing a slave or master interrupt until we we start the transfer (and know whether it's read or write)? I see the benefit of the change but is this also fixing one of the bugs you mentioned?